### PR TITLE
Update to v5.0.1

### PR DESCRIPTION
**Pull request by tadeu**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by tadeu** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -47,3 +50,4 @@ about:
 extra:
   recipe-maintainers:
     - tadeu
+    - lrineau
```
Comment (tadeu) [inline recipe/meta.yaml:57]: @lrineau is it ok to add you as maintainer?

**Comment by tadeu** [comment]
I've noticed that the license specified in `meta.yaml` is the license for CGAL ("GPL3/LGPL3"), not the license for `cgal-swig-bindings` . The license for `cgal-swig-bindings` used to be `Boost Software License`, but it seems it recently changed to a dual licensing commercial/GPL... I'm worried about having GPL code here, it might invalidate the use of this package for who is interested in just the LGPL part.

We also have to pick one `license_file`, but currently there are two.

**Comment by tadeu** [comment]
Fixes #54

**Comment by tadeu** [comment]
Failing in macOS with:

```
CMake Error in SWIG_CGAL/Kernel/CMakeLists.txt:
  No known features for CXX compiler

  "Clang"

  version 9.0.1.
```

it looks similar to this problem: https://github.com/google/cctz/issues/134

File: recipe/meta.yaml (hunk 1 above)
Comment (lrineau) [reply recipe/meta.yaml:57]: Yes, maybe. For the moment, I do not know conda very well. I will probably only review and give advice. And my colleague @maxGimeno as well.

**Comment by lrineau** [comment]
> I've noticed that the license specified in `meta.yaml` is the license for CGAL ("GPL3/LGPL3"), not the license for `cgal-swig-bindings` . The license for `cgal-swig-bindings` used to be `Boost Software License`, but it seems it recently changed to a dual licensing commercial/GPL... I'm worried about having GPL code here, it might invalidate the use of this package for who is interested in just the LGPL part.
> 
> We also have to pick one `license_file`, but currently there are two.

@sloriot You picked the new license. Which modules of our Python bindings correspond to LGPL packages of CGAL?

**Comment by lrineau** [comment]
> Failing in macOS with:
> 
> ```
> CMake Error in SWIG_CGAL/Kernel/CMakeLists.txt:
>   No known features for CXX compiler
> 
>   "Clang"
> 
>   version 9.0.1.
> ```
> 
> it looks similar to this problem: [google/cctz#134](https://github.com/google/cctz/issues/134)

That is because the `CMakeLists.txt` of cgal-swig-bindings starts with `cmake_minimum_required(VERSION 2.8)` and does not enable [`CMP0025`](https://cmake.org/cmake/help/latest/policy/CMP0025.html). That is an error: the minimal CMake version should be at least 3.1 (because CGAL itself requires 3.1), and that minimal version of CMake enables `CMP0025` by default.

https://github.com/CGAL/cgal-swig-bindings/blob/e14208e0f076c03a5c8ac69b430dad43f38c7ab4/CMakeLists.txt#L1-L22

Can a Conda package carry a patch?

**Comment by lrineau** [comment]
See also https://gitlab.kitware.com/cmake/cmake/issues/19250, that helped me understand the issue.

**Comment by sloriot** [comment]
> > I've noticed that the license specified in `meta.yaml` is the license for CGAL ("GPL3/LGPL3"), not the license for `cgal-swig-bindings` . The license for `cgal-swig-bindings` used to be `Boost Software License`, but it seems it recently changed to a dual licensing commercial/GPL... I'm worried about having GPL code here, it might invalidate the use of this package for who is interested in just the LGPL part.
> > We also have to pick one `license_file`, but currently there are two.
> 
> @sloriot You picked the new license. Which modules of our Python bindings correspond to LGPL packages of CGAL?

To keep it simple (for me), I've put all under GPL. We can readd LGPL but I guess it would be only for the Kernel package so probably not useful anyway.

**Comment by tadeu** [comment]
> Can a Conda package carry a patch?

Yes! I'll add the patch here.

**Comment by nicoddemus** [comment]
> We also have to pick one license_file, but currently there are two.

Actually it seems you can have multiple license files:

https://github.com/conda/conda-build/pull/3600

**Comment by tadeu** [comment]
Thanks @nicoddemus!

So, @lrineau and @sloriot, now there's only the `license_file` issue that we need to solve.

Since it's possible to add more than one license to the package, do you think it would be possible to also add LGPL to `cgal-swig-bindings` so that we add the three license files here, and whoever uses only the LGPL part would still be able to continue using it without a bad surprise?

I'm seeing that there are also a bunch of non-kernel packages that are LGPL here: https://doc.cgal.org/latest/Manual/packages.html

**Comment by lrineau** [comment]
With only LGPL CGAL packages, then only the `_CGAL_Kernel.so` Python extension builds. That is very limited. I think it is not useful to add the LGPL for that single module. Let's keep GPLv3+ for the whole cgal-swig-bindings software.

**Pull request merged**